src/v/cloud_roles/tests/env_override_tests.cc

```diff
@@ -11,11 +11,12 @@
 #include "cloud_roles/refresh_credentials.h"
 
 #include <seastar/core/gate.hh>
+#include <seastar/testing/thread_test_case.hh>
 #include <seastar/util/defer.hh>
 
 #include <boost/test/unit_test.hpp>
 
-BOOST_AUTO_TEST_CASE(test_override_address) {
+SEASTAR_THREAD_TEST_CASE(test_override_address) {
```

Comment:
You can't just change this, mixing `BOOST_AUTO_TEST_CASE` and `SEASTAR_THREAD_TEST_CASE` in the same binary results in one set of them not running, I think.

Reply:
they were already mixed before this in the same binary. lemme run it and see if it is working or not

Reply:
Looks like you may have actually fixed it!

Reply:
@BenPope i remember this being a thing too. but it does seem to work. there were changes iiuc to testing in seastar so maybe it was improved?

here is the experiment:

```
[nwatkins@panda clang]$ ctest --verbose -R test_cloud_roles_rpunit 2>&1 | grep warning
104: /home/nwatkins/src/redpanda/src/v/cloud_roles/tests/role_client_tests.cc(31): warning: in "test_simple_token_request": fixture test
104: /home/nwatkins/src/redpanda/src/v/cloud_roles/tests/credential_tests.cc(18): warning: in "test_gcp_headers": boost auto test


[nwatkins@panda clang]$ git diff
diff --git a/src/v/cloud_roles/tests/credential_tests.cc b/src/v/cloud_roles/tests/credential_tests.cc
index af4a2a54a..9d0e63b60 100644
--- a/src/v/cloud_roles/tests/credential_tests.cc
+++ b/src/v/cloud_roles/tests/credential_tests.cc
@@ -15,6 +15,7 @@
 namespace bh = boost::beast::http;
 
 BOOST_AUTO_TEST_CASE(test_gcp_headers) {
+    BOOST_WARN_MESSAGE(false, "boost auto test");
     auto applier = cloud_roles::make_credentials_applier(
       cloud_roles::gcp_credentials{
         .oauth_token = cloud_roles::oauth_token_str{"a-token"}});
@@ -71,4 +72,4 @@ BOOST_AUTO_TEST_CASE(test_aws_headers) {
           h.at("x-amz-content-sha256"),
           "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855");
     }
-}
\ No newline at end of file
+}
diff --git a/src/v/cloud_roles/tests/role_client_tests.cc b/src/v/cloud_roles/tests/role_client_tests.cc
index 15b7c6909..ef61f6f8e 100644
--- a/src/v/cloud_roles/tests/role_client_tests.cc
+++ b/src/v/cloud_roles/tests/role_client_tests.cc
@@ -28,6 +28,7 @@ namespace ba = boost::algorithm;
 static const cloud_roles::aws_region_name region{""};
 
 FIXTURE_TEST(test_simple_token_request, http_imposter_fixture) {
+    BOOST_WARN_MESSAGE(false, "fixture test");
     when()
       .request(cloud_role_tests::gcp_url)
       .then_reply_with(cloud_role_tests::gcp_oauth_token);

```